libs/wingc/src/lib.rs

```diff
@@ -156,6 +156,23 @@ pub unsafe extern "C" fn wingc_malloc(size: usize) -> *mut u8 {
 	}
 }
 
+/// Check if the project has a pnpm or yarn or bun lockfile, and emit a warning if it does.
+fn emit_warning_for_unsupported_package_managers(project_dir: &Utf8Path) {
```

Comment:
Inverting the check for the success case (look for a `package-lock.json` instead) could be useful since there's only one case that is supposed to work and many cases that might not.

A `package-lock.json` file may not be present in cases where there are no dependencies, but ideally we shouldn't even be checking this in that case.

Reply:
> could be useful since there's only one case that is supposed to work and many cases that might not

I guess I'm OK with the possibility that this list might become outdated (since it's easy to add onto). Like you said, if the user has no dependencies (or maybe they just aren't saving lock files), package-lock.json might not be present so I'm not sure how we'd avoid the check otherwise.

Reply:
It's not only about outdated cases, but cases this doesn't handle like workspaces since the lockfile would only present at the root, not necessarily next to a package.json

Reply:
Mmm I see. Yeah, fair enough, we can probably do more here (maybe do a lookup for the project's node_modules and check if there's a .pnpm folder, or only raise the error when a module lookup error occurs like you suggested) -- definitely open to improving it.

Let's merge this in for now, if you wanna try and fine tune the warning then I'll totes take the contribution